### ISSUE TITLE
`gen-json-schema`: Relative imports broken (at least for import map)

### ISSUE THREAD

**Issue by joeflack4** [linkml/linkml#682]
**Describe the bug**
When using `gen-json-schema` (I think only this generator, but maybe also a problem for the other ones too) with import map, it doesn't understand that the path in to the import map is relative to the project root directory.

@hsolbrig Can you check that I accurately described this issue?

**To Reproduce**
1. Open `linkml_model/model/schema/meta.yaml`
2. Add prefix: `tccm: https://hotecosystem.org/tccm/`
3. Add imports: "- tccm:valuesetcatalogentry"
4. Run `gen-json-schema` and see that it fails on HTTP error 404 not found: https://hotecosystem.org/tccm/valuesetcatalogentry.yaml
5. Add `import_map.json` to project root dir: `{"tccm:valuesetcatalogentry": "valuesetcategoryentry.yaml"}`
6. Run again with `--importmap import_map.json`, and it still fails.

**Comment by cmungall** [comment]
What is the error message?

Note it's OK to make a draft PR on linkml-model that demonstrates the issue

I am pretty sure that the reason for this is that there is nothing here:

https://hotecosystem.org/tccm/valuesetcatalogentry.yaml

@hsolbrig are you sure we want to use https://hotecosystem.org and not a w3id?

Is the intent that import_map.json serves the analog purpose of catalog.xml in OWL - is this documented somewhere?

**Comment by joeflack4** [comment]
@cmungall It is indeed as you mentioned; that that particular url did not resolve. I was watching Harold code this; his LinkML understanding is better than mine. It was my understanding that (5) and (6) were supposed to alleviate the this problem (4). Basically that the idea of `--importmap` is that you can resolve imports to local files, rather than to the web.

In order to resolve the issue, he had to move the yaml file out of the project root and into some other folder. @hsolbrig understands slightly better than I what exactly happened.

**Comment by sierra-moxon** [comment]
@joeflack4  - is this still an issue for you?

**Comment by joeflack4** [comment]
Unless anyone worked on this, I'm gonna guess it's still broken. Bit strapped on time to try to reproduce this. This was something that came up during a short pair programming session with Harold Solbrig. I tried to locate the branch we worked on, but can't find it. Maybe we ditched the work. I haven't actually seen Harold since then. This issue is not a hold up for me in any way, but it does appear to be a bug. Harold has a better grasp of linkml relative imports than I.
